### ISSUE TITLE
Azure ID Error

### ISSUE THREAD

**Issue by mosnicholas** [supabase/gotrue#1274]
# Bug report

As of `2023-10-17T08:59:15.000Z`, I have started seeing the following errors in my Azure auth:

```
azure: ID token issuer \"https://login.microsoftonline.com/TENANT-UUID/v2.0\" does not match expected issuer \"https://login.microsoftonline.com/organizations/v2.0\"","level":"error","method":"GET","msg":"500: Error getting user email from external provider"
```

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Azure login in is no longer working for my multitenant app. It might be related to this: https://github.com/supabase/gotrue/pull/1269.

## To Reproduce

I'm not sure -- i haven't changed anything in my app since yesterday and started seeing these errors.

## Expected behavior

There should be no errors.

## System information

- OS: Running in Supabase cloud.
- Version of supabase-js: ^2.36.0
- Version of Node.js: 18.2

**Comment by aaron-xandria** [comment]
I see the same thing. I think its has to do with appending"2.0" to the issuer:
https://github.com/supabase/gotrue/blob/4c50357841c51c2da0eff4d7f8920aed5e640df2/internal/api/provider/azure.go#L62

Docs state to omit that for Active directory 1.0 apps:
![unnamed (1)](https://github.com/supabase/gotrue/assets/129017605/0e95818b-18fe-46bc-b9a0-2a9ed92ebc4c)

Our app has been down for our users, hopefully this gets resolved soon.

**Comment by mosnicholas** [comment]
@hf was there a deploy for Supabase auth related to the changes you made yesterday? I don't know where to check, but I saw you merged in some changes to gotrue that seem related?

**Comment by mosnicholas** [comment]
It seems we're hitting this error: https://github.com/supabase/gotrue/pull/1269/files#diff-ea3c11f35c15f1496340532424b81c805cc3f6fb541800add5cd6f497548b324R130. I believe the changes here should have fixed it: https://github.com/supabase/gotrue/pull/1272/files. But I'm not sure how gotrue / supabase deploys work, or if that code is in production. If it is, I'm not sure why I'm still seeing this error. 

@hf given this is affecting production customers, it would be great to know if your team can act on this asap.

**Comment by aaron-xandria** [comment]
@mosnicholas not sure if you situation allows, but I was able to resolve my issue by switching my from Azure Tenant URL from:

https://login.microsoftonline.com/organizations to https://login.microsoftonline.com/common see the following for info:

https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-client-application-configuration

I was also required to add email to the login scope:

... = await supabase.auth.signInWithOAuth({
            ...
            options: {
                scopes: 'email,profile',
..
            },
        });

**Comment by mosnicholas** [comment]
Thank you for updating here @aaron-xandria

**Comment by kangmingtay** [comment]
hey everyone, we've fixed this issue in #1275 and will roll it out to all affected projects on Supabase.
